src/witan/population/england/snpp_2022.clj

```diff
@@ -0,0 +1,365 @@
+   `:calendar-year` and `:academic-year` added.
+   Dataset can be filtered by specifying (optional) values for:
+   - Upper Tier LA: via code (string) `UTLA22CD` or name (string) `UTLA22NM`,
+     or (for backwards compatibility) via name (string) `la-name`.
+   - NCYs: via (integer) `min-academic-year` and/or `max-academic-year`. 
+   - `:calendar-year`s: via (integer) `min-calendar-year` and/or `max-calendar-year`,
+     or (for backwards compatibility) via (integer) `max-year`."
+  ;; Note that compared to `witan.population.england.snpp-2018/snpp-2018->witan-send-population`
+  ;; `:calendar-year` differs by +1 and `:academic-year` by +1,
+  ;; due to consideration of SNPPs as mid-year estimates.
+  [ds LTLA22->UTLA22 & {:keys [UTLA22CD UTLA22NM la-name
+                               min-academic-year max-academic-year
+                               min-calendar-year max-calendar-year max-year]
+                        :or   {min-academic-year default-min-academic-year
+                               max-academic-year default-max-academic-year}}]
+  (-> ds
+      ;; Derive `:academic-year` from `:age-group`:
+      ;; - The SNPPs are mid-year estiamtes.
+      ;; - Therefore the age is (almost) the age on 31st August, i.e. at the start of the school year.
+      ;; - Per https://www.gov.uk/national-curriculum, 
+      ;;   children aged 5 at the start of the school year should be in NCY 1.
+      ;; - Thus the offset between age at the start of the school year and NCY is -4.
+      (tc/drop-rows (comp #{"90 and over" "All ages"} :age-group))
+      (tc/map-columns :age :int8 [:age-group] parse-long)
+      (tc/map-columns :academic-year :int8 [:age] #(- % 4))
+      ;; Select required `:academic-year`s (if specified)
+      (cond->
+       min-academic-year (tc/select-rows #(-> % :academic-year (>= min-academic-year)))
+       max-academic-year (tc/select-rows #(-> % :academic-year (<= max-academic-year))))
+      ;; Merge in Upper Tier LA codes and names
+      (tc/left-join (tc/select-columns LTLA22->UTLA22 [:LTLA22CD :LTLA22NM
+                                                       :UTLA22CD :UTLA22NM])
+                    {:left  [:area-code]
+                     :right [:LTLA22CD]})
+      (tc/reorder-columns [:area-code :area-name :LTLA22CD :LTLA22NM :UTLA22CD :UTLA22NM])
+      ;; Select Upper Tier LA (if specified)
+      (cond->
+       UTLA22CD (tc/select-rows #(-> % :UTLA22CD (= UTLA22CD)))
+       UTLA22NM (tc/select-rows #(-> % :UTLA22NM (= UTLA22NM)))
+       la-name  (tc/select-rows #(-> % :UTLA22NM (= la-name))))
+      ;; Pivot long with SNPP year in `:snpp-year` and projections in `:population`
+      (tc/pivot->longer #"^\d+$" {:target-columns :snpp-year, :value-column-name :population})
+      ;; Roll-up to Upper Tier LA level
+      (as-> $ (tc/group-by $ (tc/column-names $ (complement #{:area-code :area-name :LTLA22CD :LTLA22NM :population}))))
+      (tc/aggregate {:population #(-> % :population tcc/reduce-+)})
```

Comment:
I don't know why I like it when we use tcc things, but I just do

Reply:
Here it's just a wrapper for `dfn/reduce-+`, but I find the `tcc` versions easier to find. Could have used `tc/reduce-+` as an aggregator but they always put the calculated values ina column called "summary" that you then have to rename.